hail/src/main/scala/is/hail/rvd/AbstractRVDSpec.scala

```diff
@@ -25,11 +25,11 @@ object RVDSpec {
     new TStructSerializer +
     new RVDTypeSerializer
 
-  def read(hc: HailContext, path: String): RVDSpec = {
+  def read(hc: HailContext, path: String): AbstractRVDSpec = {
     val metadataFile = path + "/metadata.json.gz"
     val jv = hc.hadoopConf.readFile(metadataFile) { in => JsonMethods.parse(in) }
-        .transformField { case ("orvdType", value) => ("rvdType", value) }
-    jv.extract[RVDSpec]
+
+    Compatibility.extractRVD(jv)
```

Comment:
The mainline code shouldn't call into the compatibility layer like this.  You should be able to delete the compat package and straightforward references to it, and everything should just work.  So I'd leave UnpartitionedRVDSpec in the type hints and that's it.